src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsCollection.cs

```diff
@@ -0,0 +1,303 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Android.Runtime;
+using Java.Interop.Tools.Cecil;
+using Java.Interop.Tools.Diagnostics;
+using Java.Interop.Tools.JavaCallableWrappers;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks;
+
+class MarshalMethodsCollection : JavaCallableMethodClassifier
+{
+	/// <summary>
+	/// Assemblies that contain convertible marshal methods. These assemblies need to
+	/// have the proper assembly references added to them.
+	/// </summary>
+	public HashSet<AssemblyDefinition> AssembliesWithMarshalMethods { get; } = [];
```

Comment:
We don't put this object in `RegisterTaskObject()` since https://github.com/dotnet/android/pull/10058?

So, should be ok to store an `AssemblyDefinition` here.

Reply:
These stored `AssemblyDefinition` objects were simply moved from `MarshalMethodsClassifier`, so it should be functionally the same as what we have today.  I suspect when the assembly resolver gets disposed it will close all files held by `AssemblyDefinition`.  Attempting to use this object after that would probably result in `ObjectDisposedException` but I guess we have structured our code so we don't use these after the resolver has been disposed.